src/solver/optimisation/ConstraintGroup.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include "sim_structure_probleme_economique.h"
+#include "opt_structure_probleme_a_resoudre.h"
+#include "new_constraint_builder_utils.h"
+
+class ConstraintGroup
+{
+public:
+    explicit ConstraintGroup(PROBLEME_HEBDO* problemeHebdo) :
+     problemeHebdo_(problemeHebdo),
+     builder_(NewGetConstraintBuilderFromProblemHebdo(problemeHebdo_))
+    {
+    }
+
+    /*TODO Rename this*/
+    virtual void Build() = 0;
+    PROBLEME_HEBDO* problemeHebdo_;
+    std::shared_ptr<NewConstraintBuilder> builder_;
```

Comment:
- why shared ownership, since it's clearly owned by this object ?
- why heap allocation in the first place ? ^^

Reply:
you are right, in the first version of the code the "builder" wasn't a member object and was built for every constraint group.
Since they share the same base i couldn't agree more with you. To be done